CondTools/CTPPS/src/CTPPSRPAlignmentInfoAnalyzer.cc

```diff
@@ -72,20 +72,25 @@ CTPPSRPAlignmentInfoAnalyzer::CTPPSRPAlignmentInfoAnalyzer(const edm::ParameterS
 //----------------------------------------------------------------------------------------------------
 
 void CTPPSRPAlignmentInfoAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
-  edm::ESHandle<CTPPSRPAlignmentCorrectionsData> alignments;
+  const auto alignments = [&r = record_,
+                           &eS = iSetup,
+                           &tAI = tokenAlignmentIdeal_,
+                           &tAR = tokenAlignmentReal_,
+                           &tAM = tokenAlignmentMisaligned_]() -> const CTPPSRPAlignmentCorrectionsData* {
+    if (r == "CTPPSRPAlignmentCorrectionsDataRcd") {
+      return &eS.getData(tAI);
```

Comment:
I dont really understand why this couldnt just be
```suggestion
      return &eS.getData(tokenAlignmentIdeal_);
```

Reply:
Because, in my understanding, tAI is the captured variable. In fact, if I do your change I get compilation errors.